File: docs/public/specs/esm/window-frame.js
import * as vg from "@uwdata/vgplot";

await vg.coordinator().exec([
  vg.loadParquet("aapl", "data/stocks.parquet", {where: "Symbol = 'AAPL'"})
]);

export default vg.plot(
  vg.lineY(
    vg.from("aapl"),
    {stroke: "#ccc", x: "Date", y: "Close"}
  ),
  vg.lineY(
    vg.from("aapl"),
    {
      stroke: "black",
      x: "Date",
      y: vg.avg("Close").orderby("Date").frame(vg.frameRange([vg.days(15), vg.days(15)]))
    }
  ),
  vg.lineY(
    vg.from("aapl"),
    {
      stroke: "firebrick",
      x: "Date",
      y: vg.avg("Close").orderby("Date").frame(vg.frameRange([vg.months(3), vg.months(3)]))
    }
  ),
  vg.yLabel("Close"),
  vg.width(680),
  vg.height(200)
);

File: docs/public/specs/python/window-frame.py
import json
import vgplot as vg

data = vg.data(
    aapl={
    "type": "parquet",
    "file": "data/stocks.parquet",
    "where": "Symbol = 'AAPL'"
}
)

view = vg.plot(
    vg.line_y(data=vg.from_("aapl"), stroke="#ccc", x="Date", y="Close"),
    vg.line_y(data=vg.from_("aapl"), stroke="black", x="Date", y={
        "avg": "Close",
        "orderby": "Date",
        "range": [
        {
        "days": 15
    },
        {
        "days": 15
    }
    ]
    }),
    vg.line_y(data=vg.from_("aapl"), stroke="firebrick", x="Date", y={
        "avg": "Close",
        "orderby": "Date",
        "range": [
        {
        "months": 3
    },
        {
        "months": 3
    }
    ]
    }),
    vg.y_label("Close"),
    vg.width(680),
    vg.height(200)
)

spec = vg.spec(data=data, view=view)

if __name__ == "__main__":
    print(json.dumps(spec.to_dict(), sort_keys=True))